plonkish_backend/src/accumulation/protostar/ivc/aggregation/agg_circuit.rs

```diff
@@ -154,153 +155,224 @@ impl CircuitExt<grumpkin::Fr> for SecondaryAggregationCircuit {
-
-//             let h = chip.fit_base_in_scalar(
-//                 &secondary_aggregation_instance[1 + 2 * self.secondary_arity],
-//             )?;
-//             chip.constrain_equal(&mut pool, &h_ohs_from_last_nark, &h)?;
-
-//             let iter = &mut secondary_aggregation_instance.iter();
-//             let mut instances = |skip: usize, take: usize| {
-//                 iter.skip(skip)
-//                     .take(take)
-//                     .map(|base| chip.to_repr_base(base))
-//                     .try_collect::<_, Vec<_>, _>()
-//             };
-//             (
-//                 instances(1, self.secondary_arity)?,
-//                 instances(0, self.secondary_arity)?,
-//                 instances(1, 4 * strawman::NUM_LIMBS)?,
-//             )
-//         };
-
-//         // let cell_map = chip.layout_and_clear(&mut layouter)?;
-//         // for (idx, witness) in chain![
-//         //     [primary_num_steps],
-//         //     primary_initial_input,
-//         //     primary_output,
-//         //     secondary_initial_input.into_iter().flatten(),
-//         //     secondary_output.into_iter().flatten(),
-//         //     pairing_acc.into_iter().flatten(),
-//         // ]
-//         // .enumerate()
-//         // {
-//         //     layouter.constrain_instance(cell_map[&witness.id()].cell(), chip.instance, idx)?;
-//         // }
-
-//         Ok(())
-//     }
-// }
-
-// impl CircuitExt<bn256::Fr> for PrimaryAggregationCircuit {
-//     fn instances(&self) -> Vec<Vec<bn256::Fr>> {
-//         vec![self.instances.clone()]
-//     }
-// }
\ No newline at end of file
+    //todo fix this with other synthesizes
+    fn synthesize(
+        &self,
+        config: Self::Config,
+        mut layouter: impl Layouter<bn256::Fr>,
+    ) -> Result<(), Error> {
+
+        let mut builder = RangeCircuitBuilder::from_stage(CircuitBuilderStage::Keygen)
+        .use_k(8)
+        .use_lookup_bits(9);
+
+        let range = builder.range_chip();
+        let gate_chip = GateChip::<bn256::Fr>::new();
+        let base_chip = FpChip::<bn256::Fr, bn256::Fq>::new(&range, NUM_LIMB_BITS, NUM_LIMBS);
+        let native_chip = NativeFieldChip::new(&range);
+        // Error Correcting Chip 
+        let ecc_chip = EccChip::new(&native_chip);
+
+        let mut pool = mem::take(builder.pool(0));
+        
+        // Chip here is built with a combination of chips ß
+        /*
+        
+            Initially passing three arguments, but requires only a range chip. 
+            let range passed
+         */
+        let chip = strawman::Chip::<bn256::G1Affine>::create(range);
+
+        // let chip =
+        //     <strawman::Chip<bn256::G1Affine> as TwoChainCurveInstruction<bn256::G1Affine>>::new(chip,
+        //         config,
+        //     );
+
+        let mut builder = RangeCircuitBuilder::from_stage(CircuitBuilderStage::Keygen)
```

Comment:
why there is a need for two builders here? look how to intialise a builder in fn synthesize_accumulation_verifier